### PR TITLE
Cloud9 integration for Conditions

### PR DESCRIPTION
**Pull request by snjkumar23**
## What does this PR do and why?

Cloud9 Changes for Conditions sections

## PR acceptance checklist

This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [] Added/Updated documentation if applicable.
- [x] Pre-commit checks passed.
- [x] Lint and Nag checks passed.
- [ ] If releasing a new version, have you bumped the version `make version part=<major|minor|patch>`?

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: content/intermediate/templates/conditions/index.md [hunk 1]
```diff
@@ -109,31 +109,76 @@ Let’s deploy the solution!
...
-* Pass `test` as the value for the `EnvType` parameter. Choose **Next**.
-* Choose to accept default values in the **Configure stack options** page. Choose **Next**.
-* Choose **Create stack**. You can view the progress of the stack being created in the CloudFormation console.
-* Wait until the stack creation is complete. Refresh the view in the console until you see your stack to be in the `CREATE_COMPLETE` status.
-
+:::::tabs{variant="container"}
+::::tab{id="cloud9" label="Cloud9"}
+1. Upload the file to your **template S3 bucket** using AWS CLI [aws s3 cp](https://docs.aws.amazon.com/cli/latest/reference/s3/cp.html) command.
+
+   `aws s3 cp code/workspace/condition-resource.yaml s3://cfn-workshop-01-{accountid}`
```
Comment (cjhar) [inline content/intermediate/templates/conditions/index.md:116]: Can we please use code blocks to make it clearer and neater?
`:::code{language=shell showLineNumbers=false showCopyAction=true}
aws s3 cp code/workspace/condition-resource.yaml s3://cfn-workshop-01-{accountid}
:::`

File: content/intermediate/templates/conditions/index.md [hunk 2]
```diff
@@ -109,31 +109,76 @@ Let’s deploy the solution!
...
+
+1. Determine the **Object URL** as you'll need it in the next step, based on this format `https://[bucketname].s3.amazonaws.com/[key]`
+    for example:-
+     `https://cfn-workshop-01-{accountid}.s3.amazonaws.com/condition-resource.yaml`
```
Comment (cjhar) [inline content/intermediate/templates/conditions/index.md:120]: comment block
`:::code{language=shell showLineNumbers=false showCopyAction=true}
https://cfn-workshop-01-{accountid}.s3.amazonaws.com/{template-name}.yaml
:::`

File: content/intermediate/templates/conditions/index.md [hunk 3]
```diff
@@ -235,15 +303,37 @@ Outputs:
...
-* `EnvType` should already be set to `prod`. Choose **Next**.
-* Choose to accept default values in the **Configure stack options** page. Choose **Next**.
-* Choose **Update stack**. You can view the progress of the stack being created in the CloudFormation console.
-* Wait until the stack creation is complete. Refresh the view in the console until you see your stack to be in the `UPDATE_COMPLETE` status.
-
+  :::::tabs{variant="container"}
+    ::::tab{id="cloud9" label="Cloud9"}
+    1. Upload the file to your **template S3 bucket** using AWS CLI [aws s3 cp](https://docs.aws.amazon.com/cli/latest/reference/s3/cp.html) command.
+
+        `aws s3 cp code/workspace/condition-resource.yaml s3://cfn-workshop-01-{accountid}`
```
Comment (cjhar) [inline content/intermediate/templates/conditions/index.md:310]: comment block

File: content/intermediate/templates/conditions/index.md [hunk 4]
```diff
@@ -235,15 +303,37 @@ Outputs:
...
+
+    1. Determine the **Object URL** as you'll need it in the next step, based on this format `https://[bucketname].s3.amazonaws.com/[key]`
+        for example:-
+        `https://cfn-workshop-01-{accountid}.s3.amazonaws.com/condition-resource.yaml`
```
Comment (cjhar) [inline content/intermediate/templates/conditions/index.md:314]: comment block

File: content/intermediate/templates/conditions/index.md [hunk 5]
```diff
@@ -190,15 +235,38 @@ Time to deploy your resources!
...
-* Wait until the stack creation is complete. Refresh the view in the console until you see your stack to be in the `CREATE_COMPLETE` status.
+:::::tabs{variant="container"}
+::::tab{id="cloud9" label="Cloud9"}
+1. Upload the file to your **template S3 bucket** using AWS CLI [aws s3 cp](https://docs.aws.amazon.com/cli/latest/reference/s3/cp.html) command.
+
+   `aws s3 cp code/workspace/condition-resource.yaml s3://cfn-workshop-01-{accountid}`
+
+1. Determine the **Object URL** as you'll need it in the next step, based on this format `https://[bucketname].s3.amazonaws.com/[key]`
+    for example:-
+     `https://cfn-workshop-01-{accountid}.s3.amazonaws.com/condition-resource.yaml`
```
Comment (cjhar) [inline content/intermediate/templates/conditions/index.md:246]: comment block

File: content/intermediate/templates/conditions/index.md [hunk 6]
```diff
@@ -190,15 +235,38 @@ Time to deploy your resources!
...
-* Enter a **Stack name**. For example, choose to specify `cfn-workshop-condition-property-test`.
-* Pass `test` as the value for the `EnvType` parameter. Choose **Next**.
-* Choose to accept default values in the **Configure stack options** page. Choose **Next**.
-* Choose **Create stack**. You can view the progress of the stack being created in the CloudFormation console.
-* Wait until the stack creation is complete. Refresh the view in the console until you see your stack to be in the `CREATE_COMPLETE` status.
+:::::tabs{variant="container"}
+::::tab{id="cloud9" label="Cloud9"}
+1. Upload the file to your **template S3 bucket** using AWS CLI [aws s3 cp](https://docs.aws.amazon.com/cli/latest/reference/s3/cp.html) command.
+
+   `aws s3 cp code/workspace/condition-resource.yaml s3://cfn-workshop-01-{accountid}`
```
Comment (cjhar) [inline content/intermediate/templates/conditions/index.md:242]: comment block

File: content/intermediate/templates/conditions/index.md [hunk 7]
```diff
@@ -109,31 +109,76 @@ Let’s deploy the solution!
...
-* Enter a **Stack name**. For example, choose to specify `cfn-workshop-condition-prod`.
-* Pass `prod` as the value for the `EnvType` parameter. Choose **Next**.
-* Choose to accept default values in the **Configure stack options** page. Choose **Next**.
-* Choose **Create stack**. You can view the progress of the stack being created in the CloudFormation console.
-* Wait until the stack creation is complete. Refresh the view in the console until you see your stack to be in the `CREATE_COMPLETE` status.
+:::::tabs{variant="container"}
+::::tab{id="cloud9" label="Cloud9"}
+1. Upload the file to your **template S3 bucket** using AWS CLI [aws s3 cp](https://docs.aws.amazon.com/cli/latest/reference/s3/cp.html) command.
+
+   `aws s3 cp code/workspace/condition-resource.yaml s3://cfn-workshop-01-{accountid}`
```
Comment (cjhar) [inline content/intermediate/templates/conditions/index.md:154]: comment block

File: content/intermediate/templates/conditions/index.md (hunk 1 above)
Comment (cjhar) [inline content/intermediate/templates/conditions/index.md:116]:
```suggestion
:::code{language=shell showLineNumbers=false showCopyAction=true}
aws s3 cp code/workspace/condition-resource.yaml s3://cfn-workshop-01-{accountid}
:::
```

File: content/intermediate/templates/conditions/index.md (hunk 2 above)
Comment (cjhar) [inline content/intermediate/templates/conditions/index.md:120]:
```suggestion
:::code{language=shell showLineNumbers=false showCopyAction=true}
https://cfn-workshop-01-{accountid}.s3.amazonaws.com/condition-resource.yaml
:::
```

File: content/intermediate/templates/conditions/index.md (hunk 5 above)
Comment (cjhar) [inline content/intermediate/templates/conditions/index.md:246]:
```suggestion
:::code{language=shell showLineNumbers=false showCopyAction=true}
https://cfn-workshop-01-{accountid}.s3.amazonaws.com/condition-resource.yaml
:::
```

File: content/intermediate/templates/conditions/index.md (hunk 4 above)
Comment (cjhar) [inline content/intermediate/templates/conditions/index.md:314]:
```suggestion
:::code{language=shell showLineNumbers=false showCopyAction=true}
https://cfn-workshop-01-{accountid}.s3.amazonaws.com/condition-resource.yaml
:::
```

File: content/intermediate/templates/conditions/index.md [hunk 8]
```diff
@@ -109,31 +109,68 @@ Let’s deploy the solution!
...
+::::
+::::tab{id="local" label="Local development"}
+1. In the CloudFormation console, select **Create stack**, **With new resources (standard)**.
+1. In **Prepare template**, select **Template is ready**.
+1. In **Template source**, select **Upload a template file**.
+1. Choose the `condition-resource.yaml` template.
+1. Enter a **Stack name**. For example, choose to specify `cfn-workshop-condition-test`.
+1. Pass `test` as the value for the `EnvType` parameter. Choose **Next**.
+1. Choose to accept default values in the **Configure stack options** page. Choose **Next**.
+1. Choose **Create stack**. You can view the progress of the stack being created in the CloudFormation console.
```
Comment (sdangol) [inline content/intermediate/templates/conditions/index.md:136]:
```suggestion
1. Choose **Submit**. You can view the progress of the stack being created in the CloudFormation console.
```

File: content/intermediate/templates/conditions/index.md
Comment (sdangol) [inline content/intermediate/templates/conditions/index.md:1]: Indentation issue in line [286 - 289](https://github.com/aws-samples/cfn101-workshop/pull/210/files#diff-8487daf746d800371c656673b543b8d5a4fb4e18415b40284a4ca07b0b2053f3R286-R289)

File: content/intermediate/templates/conditions/index.md [hunk 9]
```diff
@@ -235,19 +291,37 @@ Outputs:
...
+  :::code{language=shell showLineNumbers=false showCopyAction=true}
+  cd cfn101-workshop/code/workspace/conditions
+  :::
+1. Use the AWS CLI to update the stack. The required parameter `--template-body` have been pre-filled for you.
+  :::code{language=shell showLineNumbers=false showCopyAction=true}
+  aws cloudformation update-stack --stack-name cfn-workshop-condition-prod --template-body file://condition-resource.yaml --parameters ParameterKey="EnvType",ParameterValue="prod"
+  :::
+1. If the `update-stack` command was successfully sent, CloudFormation will return `StackId`.
+  :::code{language=shell showLineNumbers=false showCopyAction=false}
+  "StackId": "arn:aws:cloudformation:us-east-1:123456789012:stack/cfn-workshop-prod/739fafa0-e4d7-11ed-a000-12d9009553ff"
```
Comment (sdangol) [inline content/intermediate/templates/conditions/index.md:306]:
```suggestion
  "StackId": "arn:aws:cloudformation:us-east-1:123456789012:stack/cfn-workshop-condition-prod/739fafa0-e4d7-11ed-a000-12d9009553ff"
```

File: content/intermediate/templates/conditions/index.md [hunk 10]
```diff
@@ -235,19 +291,37 @@ Outputs:
...
+1. Open the **[AWS CloudFormation](https://console.aws.amazon.com/cloudformation)** console in a new tab and check if the stack status is **UPDATE_COMPLETE**.
+::::
+::::tab{id="local" label="Local development"}
+1. In the CloudFormation console, select **Update stack**.
+1. In **Prepare template**, select **Replace current template**.
+1. In **Template source**, select **Upload a template file**.
+1. Choose the `condition-resource.yaml` template.
+1. `EnvType` should already be set to `prod`. Choose **Next**.
+1. Choose to accept default values in the **Configure stack options** page. Choose **Next**.
+1. Choose **Update stack**. You can view the progress of the stack being created in the CloudFormation console.
```
Comment (sdangol) [inline content/intermediate/templates/conditions/index.md:316]:
```suggestion
1. Choose **Submit**. You can view the progress of the stack being created in the CloudFormation console.
```

**Review by sdangol — CHANGES_REQUESTED** [review]
Please have a look at the comments

File: content/intermediate/templates/conditions/index.md [hunk 11]
```diff
@@ -109,31 +109,68 @@ Let’s deploy the solution!
...
-
+:::::tabs{variant="container"}
+::::tab{id="cloud9" label="Cloud9"}
+1. In the **Cloud9 terminal** navigate to `code/workspace/conditions`:
+:::code{language=shell showLineNumbers=false showCopyAction=true}
+cd cfn101-workshop/code/workspace/conditions
+:::
+1. Use the AWS CLI to create the stack. The required parameter `--template-body` have been pre-filled for you.
+:::code{language=shell showLineNumbers=false showCopyAction=true}
+aws cloudformation create-stack --stack-name cfn-workshop-condition-test --template-body file://condition-resource.yaml --parameters ParameterKey="EnvType",ParameterValue="test"
```
Comment (cjhar) [inline content/intermediate/templates/conditions/index.md:120]:
```suggestion
aws cloudformation create-stack --stack-name cfn-workshop-condition-test \
--template-body file://condition-resource.yaml \
--parameters ParameterKey="EnvType",ParameterValue="test"
```

File: content/intermediate/templates/conditions/index.md [hunk 12]
```diff
@@ -109,31 +109,68 @@ Let’s deploy the solution!
...
-* Choose **Create stack**. You can view the progress of the stack being created in the CloudFormation console.
-* Wait until the stack creation is complete. Refresh the view in the console until you see your stack to be in the `CREATE_COMPLETE` status.
-
+:::::tabs{variant="container"}
+::::tab{id="cloud9" label="Cloud9"}
+1. In the **Cloud9 terminal** navigate to `code/workspace/conditions`:
+:::code{language=shell showLineNumbers=false showCopyAction=true}
+cd cfn101-workshop/code/workspace/conditions
+:::
+1. Use the AWS CLI to create the stack. The required parameter `--template-body` have been pre-filled for you.
```
Comment (cjhar) [inline content/intermediate/templates/conditions/index.md:118]:
```suggestion
1. Use the AWS CLI to create the stack. The required parameters have been pre-filled for you.
```

File: content/intermediate/templates/conditions/index.md [hunk 13]
```diff
@@ -109,31 +109,68 @@ Let’s deploy the solution!
...
-* Choose to accept default values in the **Configure stack options** page. Choose **Next**.
-* Choose **Create stack**. You can view the progress of the stack being created in the CloudFormation console.
-* Wait until the stack creation is complete. Refresh the view in the console until you see your stack to be in the `CREATE_COMPLETE` status.
+:::::tabs{variant="container"}
+::::tab{id="cloud9" label="Cloud9"}
+1. In the **Cloud9 terminal** navigate to `code/workspace/conditions`:
+:::code{language=shell showLineNumbers=false showCopyAction=true}
+cd cfn101-workshop/code/workspace/conditions
+:::
+1. Use the AWS CLI to create the stack. The required parameter `--template-body` have been pre-filled for you.
```
Comment (cjhar) [inline content/intermediate/templates/conditions/index.md:152]:
```suggestion
1. Use the AWS CLI to create the stack. The required parameters have been pre-filled for you.
```

File: content/intermediate/templates/conditions/index.md [hunk 14]
```diff
@@ -109,31 +109,68 @@ Let’s deploy the solution!
...
+:::code{language=shell showLineNumbers=false showCopyAction=true}
+aws cloudformation create-stack --stack-name cfn-workshop-condition-prod --template-body file://condition-resource.yaml --parameters ParameterKey="EnvType",ParameterValue="prod"
```
Comment (cjhar) [inline content/intermediate/templates/conditions/index.md:154]:
```suggestion
aws cloudformation create-stack --stack-name cfn-workshop-condition-prod \
--template-body file://condition-resource.yaml \
--parameters ParameterKey="EnvType",ParameterValue="prod"
```

File: content/intermediate/templates/conditions/index.md [hunk 15]
```diff
@@ -190,15 +227,34 @@ Time to deploy your resources!
...
-* Choose to accept default values in the **Configure stack options** page. Choose **Next**.
-* Choose **Create stack**. You can view the progress of the stack being created in the CloudFormation console.
-* Wait until the stack creation is complete. Refresh the view in the console until you see your stack to be in the `CREATE_COMPLETE` status.
+:::::tabs{variant="container"}
+::::tab{id="cloud9" label="Cloud9"}
+1. In the **Cloud9 terminal** navigate to `code/workspace/conditions`:
+:::code{language=shell showLineNumbers=false showCopyAction=true}
+cd cfn101-workshop/code/workspace/conditions
+:::
+1. Use the AWS CLI to create the stack. The required parameter `--template-body` have been pre-filled for you.
```
Comment (cjhar) [inline content/intermediate/templates/conditions/index.md:236]:
```suggestion
1. Use the AWS CLI to create the stack. The required parameters have been pre-filled for you.
```

File: content/intermediate/templates/conditions/index.md [hunk 16]
```diff
@@ -190,15 +227,34 @@ Time to deploy your resources!
...
+:::code{language=shell showLineNumbers=false showCopyAction=true}
+aws cloudformation create-stack --stack-name cfn-workshop-condition-property-test --template-body file://condition-resource-property.yaml --parameters ParameterKey="EnvType",ParameterValue="test"
```
Comment (cjhar) [inline content/intermediate/templates/conditions/index.md:238]:
```suggestion
aws cloudformation create-stack --stack-name cfn-workshop-condition-property-test \
--template-body file://condition-resource-property.yaml \
--parameters ParameterKey="EnvType",ParameterValue="test"
```

File: content/intermediate/templates/conditions/index.md [hunk 17]
```diff
@@ -190,15 +227,34 @@ Time to deploy your resources!
...
+::::
+::::tab{id="local" label="Local development"}
+1. In the CloudFormation console, select **Create stack**, **With new resources (standard)**.
+1. In **Prepare template**, select **Template is ready**.
+1. In **Template source**, select **Upload a template file**.
+1. Choose the `condition-resource-property.yaml` template.
+1. Enter a **Stack name**. For example, choose to specify `cfn-workshop-condition-property-test`.
+1. Pass `test` as the value for the `EnvType` parameter. Choose **Next**.
+1. Choose to accept default values in the **Configure stack options** page. Choose **Next**.
+1. Choose **Create stack**. You can view the progress of the stack being created in the CloudFormation console.
```
Comment (cjhar) [inline content/intermediate/templates/conditions/index.md:254]:
```suggestion
1. Choose **Submit**. You can view the progress of the stack being created in the CloudFormation console.
```

File: content/intermediate/templates/conditions/index.md [hunk 18]
```diff
@@ -235,19 +291,37 @@ Outputs:
...
-* Choose **Update stack**. You can view the progress of the stack being created in the CloudFormation console.
-* Wait until the stack creation is complete. Refresh the view in the console until you see your stack to be in the `UPDATE_COMPLETE` status.
-
+:::::tabs{variant="container"}
+::::tab{id="cloud9" label="Cloud9"}
+1. In the **Cloud9 terminal** navigate to `code/workspace/conditions`:
+  :::code{language=shell showLineNumbers=false showCopyAction=true}
+  cd cfn101-workshop/code/workspace/conditions
+  :::
+1. Use the AWS CLI to update the stack. The required parameter `--template-body` have been pre-filled for you.
```
Comment (cjhar) [inline content/intermediate/templates/conditions/index.md:300]:
```suggestion
1. Use the AWS CLI to update the stack. The required parameters have been pre-filled for you.
```

File: content/intermediate/templates/conditions/index.md [hunk 19]
```diff
@@ -235,19 +291,37 @@ Outputs:
...
+  :::code{language=shell showLineNumbers=false showCopyAction=true}
+  aws cloudformation update-stack --stack-name cfn-workshop-condition-prod --template-body file://condition-resource.yaml --parameters ParameterKey="EnvType",ParameterValue="prod"
```
Comment (cjhar) [inline content/intermediate/templates/conditions/index.md:302]:
```suggestion
  aws cloudformation update-stack --stack-name cfn-workshop-condition-prod \
  --template-body file://condition-resource.yaml \
  --parameters ParameterKey="EnvType",ParameterValue="prod"
```

**Review by cjhar — CHANGES_REQUESTED** [review]
Thanks, content is good just some formatting and minor corrections. End to end tested okay

**Review by cjhar — COMMENTED** [review]
Thanks Sanjeev, looks good but I would break up the Challenge command on line #302
```
aws cloudformation update-stack --stack-name cfn-workshop-condition-prod \
--template-body file://condition-resource.yaml \
--parameters ParameterKey="EnvType",ParameterValue="prod"```

**Comment by snjkumar23** [comment]
> Thanks Sanjeev, looks good but I would break up the Challenge command on line #302
> 
> ```
> aws cloudformation update-stack --stack-name cfn-workshop-condition-prod \
> --template-body file://condition-resource.yaml \
> --parameters ParameterKey="EnvType",ParameterValue="prod"```
> ```

done

**Review by cjhar — COMMENTED** [review]
LGTM

**Review by cjhar — APPROVED** [review]

**Pull request merged**